docker/Dockerfile

```diff
@@ -70,9 +72,7 @@ RUN python3 -m venv $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 RUN rm -rf tmp && git clone --depth 1 --branch ${LIGHTNING_VERSION} https://github.com/PennyLaneAI/pennylane-lightning.git tmp\
     && mv tmp/* /opt/pennylane-lightning && rm -rf tmp
-RUN pip install --no-cache-dir wheel
-RUN pip install --no-cache-dir -r requirements-dev.txt
-RUN pip install --no-cache-dir -r doc/requirements.txt
+RUN pip install --no-cache-dir cmake ninja pybind11 wheel
```

Comment:
That's all we need from `requirements-dev.txt`.

Reply:
Some packages like `clang-tidy` have been a source of failures on some platform as reported by @jay-selby previously, and hence I'm removing it to favor a minimal set of dependencies. `requirements-dev.txt` has many unnecessary packages and Docker style guides usually advocates minimal stages and images. `quimb` is not really integrated in the Lightning env yet, but we can add it if/when it is.

Reply:
Sounds good! :+1:

Reply:
> Maybe prioritize robustness here since that is the goal of the story?

Agreed, but in this case I would argue adding extra requirements was decreasing robustness since some unrequired dependencies in `requirements-dev.txt` were failing the build. So I would like to keep it as is.